docs/discovery.md

```diff
@@ -155,3 +102,58 @@ class MyClass
     }
 }
 ```
+
+
+## Integrating your own implementation with the discovery mechanism
+
+The `php-http/discovery` has built-in support for some implementations. To use a different implementation or override 
+the default when several implementations are available in your codebase, you can register a class explicitly with the 
```

Comment:
matter of taste. makes commenting on longer sections harder in a diff...
but on the other hand the 100 character lines lead to very random
editing when adding something in the middle of a section. fixing.